test/test_server.c

```diff
@@ -172,12 +212,40 @@ int main_server(void)
 				printf("recv lua err\n");
 			}
 		}
+		sscanf(buf, "%s %s", method, path);
+
+		strcpy(real_path, www_path);
+		fd = open(strcat(real_path, path), O_RDONLY);
+		if (!fd) {
+			send(client, DATA_404, strlen(DATA_404), 0);
+		} else {
+			fstat(fd, &file_stat);
+			sprintf(buf, DATA_200, file_stat.st_size);
+			send(client, buf, strlen(buf), 0);
+			sendfile(client, fd, NULL, file_stat.st_size);
+			close(fd);
+		}
 #else
 		gnutls_record_recv(session, buf, sizeof(buf));
+		sscanf(buf, "%s %s", method, path);
+
+		strcpy(real_path, www_path);
+		fd = open(strcat(real_path, path), O_RDONLY);
+		if (!fd) {
+			gnutls_record_send(session, DATA_404, strlen(DATA_404));
+		} else {
+			fstat(fd, &file_stat);
+			sprintf(buf, DATA_200, file_stat.st_size);
```

Comment:
use `asprintf` instead of a static buffer, you may also cache these replies before actually receiving the `GET` requests. Please review this for kTLS as well.